sympy/sets/tests/test_fancysets.py

```diff
@@ -518,6 +603,16 @@ def test_ImageSet_simplification():
 def test_ImageSet_contains():
     from sympy.abc import x
     assert (2, S.Half) in imageset(x, (x, 1/x), S.Integers)
+    # not testing to see if an element of the set, e.g. (2, 1/2)
+    # contains S.Half, testing to see if the given element is in
+    # the set. Since all the elements are tuples, S.Half is not
+    # in the set.
+    assert S.Half not in imageset(x, (x, 1/x), S.Integers)
+    assert (1, 2, 3) not in imageset(x, (x, 1/x), S.Integers)
+    raises(ValueError, lambda:
+        {1, 2} in imageset(x, (x, 1/x), S.Integers))
+    assert (1, 2) not in imageset(x, x + 1, S.Integers)
+    assert {1} in imageset(x, {x}, S.Integers)
```

Comment:
Why should `{1, 2}` raise?

Reply:
`{1, 2}` is a set, not an Integer; `S.Integers.intersect({1,2}) == {1, 2}`, however.

Reply:
Should it not be False then since `{1, 2}` is not an element of the set? In general a set can be an element of another set so I don't see why it needs to raise

Reply:
Agree. My thinking was that we should be more careful with ImageSet otherwise we let something fail silently that didn't make sense for the query (like `Eq(I, 0) -> False vs Lt(I, 0) -> error`). In my answer, even, I got it wrong. {1, 2} fails because a set will not be in a set of tuples, not because it is not an Integer. `1 in imageset(x, (x, 1/x), S.Integers)` also fails.

Will change.